pkg/controller/service/normalizing.go

```diff
@@ -2,18 +2,58 @@ package service
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"math"
 	"reflect"
 
+	jsonpatch "github.com/evanphx/json-patch/v5"
 	"github.com/go-openapi/jsonpointer"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func IgnoreJSONPaths(ctx context.Context, c client.Client, obj unstructured.Unstructured, ignorePaths []string) (unstructured.Unstructured, error) {
+func IgnoreJSONPaths(obj unstructured.Unstructured, ignorePaths []string) (unstructured.Unstructured, error) {
+	var ops []map[string]string
+	for _, path := range ignorePaths {
+		ops = append(ops, map[string]string{
+			"op":   "remove",
+			"path": path,
+		})
+	}
+
+	patchJSON, err := json.Marshal(ops)
+	if err != nil {
+		return obj, fmt.Errorf("failed to marshal patch: %w", err)
+	}
+
+	raw, err := json.Marshal(obj.Object)
+	if err != nil {
+		return obj, fmt.Errorf("failed to marshal object: %w", err)
+	}
+
+	patch, err := jsonpatch.DecodePatch(patchJSON)
```

Comment:
is there no way we can manually create this json patch struct w/o routing it through a json representation?